bayesmark/builtin_opt/hyperopt_optimizer.py

```diff
@@ -227,7 +227,6 @@ def observe(self, X, y):
             x_guess_ = HyperoptOptimizer.hashable_dict(x_guess)
             assert x_guess_ in self.trial_id_lookup, "Appears to be guess that did not originate from suggest"
 
-            assert x_guess_ in self.trial_id_lookup, "trial object not available in trial dict"
```

Comment:
@rdturnermtl the motivation of this pr is that this assertion is already checked at L228.